chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -550,12 +550,20 @@ private[core] sealed trait UIntFactory {
   // NOTE: This resolves UInt(width = 32)
   def apply(dir: Option[Direction] = None, width: Int): UInt = apply(Width(width))
   /** Create a UInt literal with inferred width.- compatibility with Chisel2. */
-  def apply(value: BigInt): UInt = apply(value, Width())
+  def apply(value: BigInt)(implicit opts: CompileOptions): UInt = {
+    if (opts.deprecateSingleArgumentFactoryMethods) {
```

Comment:
Also, if we could make the compile options known at Scala-compile time, this could make deprecations more inline with Scala's native compile-time `@deprecated` annotation. With macro paradise (which will be pulled in with the naming annotations), this could also be an annotation.